### DISCUSSION TITLE
Tabbed layout won't restore

### DISCUSSION THREAD

**Discussion by talkshowhost85** [i3/i3#4513] [Q&A] [answered]
I'm new to i3wm (like 5 hours new). After reading here and there I managed to figure out how to make layout, save them, restore them and swallow the appropriate programs into each container.

So I start to make a layout that I think I will like. Basically two main columns:

1) left column has 1 container for Firefox (I set this at 0.7 width)
2) right column has 4 tabbed containers for Thunar, Atom, Spotify & Telegram (I set this at 0.3 width)

My problems:

1) My Spotify sometimes load and sometimes doesn't
2) My tabbed layout ends up with 5 equal-width vertical containers instead (sometimes 4, when Spotify doesn't load)
3) My programmes don't load in correct sequence according to my i3/config file > guake, firefox, thunar, atom, spotify, telegram so those vertical containers (in point 2) get jumbled up - btw I used "exec" for each programme

Initially when I used two columns with the right column having 2 hsplit containers, they loaded just fine.

Could someone point me where to read further about advanced layout restoration? Thank you :)

PS: my layout file is attached

[default.json.txt](https://github.com/i3/i3/files/7163669/default.json.txt)

**Comment by Airblader** [comment, marked as answer]
> My Spotify sometimes load and sometimes doesn't

Spotify is known to be problematic because it sets its WM_CLASS after mapping the window. Matching on the class might not work.

> My programmes don't load in correct sequence

There is no sequence to keep. i3 will execute the commands in order, but that just launches a process. If, when, and how many windows that process opens is up to the process and nothing i3 can control or guarantee. That's why matching on attributes is important.

**Reply by talkshowhost85** [reply]
Ok thank you 🙂
